pennylane/shadows/classical_shadow.py

```diff
@@ -0,0 +1,140 @@
+            qml.matrix(qml.PauliX(0)),
+            qml.matrix(qml.PauliY(0)),
+            qml.matrix(qml.PauliZ(0)),
+        ]
+
+    def local_snapshots(self, wires=None, snapshots=None):
+        r"""Compute the T x n x 2 x 2 local snapshots
+        i.e. compute :math:`3 U_i^\dagger |b_i \rangle \langle b_i| U_i - 1` for each qubit and each snapshot
+        """
+        bitstrings = self.bitstrings
+        recipes = self.recipes
+
+        if isinstance(snapshots, int):
+            pick_snapshots = np.random.choice(
+                np.arange(snapshots, dtype=np.int64), size=snapshots, replace=False
+            )
+            pick_snapshots = qml.math.convert_like(pick_snapshots, bitstrings)
+            bitstrings = qml.math.gather(bitstrings, pick_snapshots)
+            recipes = qml.math.gather(recipes, pick_snapshots)
+
+        if isinstance(wires, Iterable):
+            bitstrings = bitstrings[:, wires]
+            recipes = recipes[:, wires]
+
+        T, n = bitstrings.shape
+
+        U = np.empty((T, n, 2, 2), dtype="complex")
+        for i, u in enumerate(self.observables):
+            U[np.where(recipes == i)] = u
+
+        state = (
+            qml.math.cast((1 - 2 * bitstrings[:, :, None, None]), np.complex64) * U + np.eye(2)
+        ) / 2
+
+        return 3 * state - np.eye(2)[None, None, :, :]
+
+    @staticmethod
+    def _obtain_global_snapshots(local_snapshot):
+        global_snapshots = []
+        for T_snapshot in local_snapshot:
+            tensor_product = T_snapshot[0]
+            for n_snapshot in T_snapshot[1:]:
+                tensor_product = np.kron(tensor_product, n_snapshot)
+            global_snapshots.append(tensor_product)
+
+        return np.array(global_snapshots)
+
+    def global_snapshots(self, wires=None, snapshots=None):
+        """Compute the T x 2**n x 2**n global snapshots"""
+
+        local_snapshot = self.local_snapshots(wires, snapshots)
+
+        if local_snapshot.shape[1] > 16:
+            warnings.warn(
+                "Querying density matrices for n_wires > 16 is not recommended, operation will take a long time",
+                UserWarning,
+            )
+
+        return ClassicalShadow._obtain_global_snapshots(local_snapshot)
+
+    def _convert_to_pauli_words(self, observable):
+        """Given an observable, obtain a list of coefficients and Pauli words, the
+        sum of which is equal to the observable"""
+
+        num_wires = self.bitstrings.shape[1]
+        obs_to_recipe_map = {"PauliX": 0, "PauliY": 1, "PauliZ": 2}
+
+        def pauli_list_to_word(obs):
+            word = [-1] * num_wires
+            for ob in obs:
+                if ob.name not in obs_to_recipe_map:
+                    raise ValueError("Observable must be a linear combination of Pauli observables")
+
+                word[ob.wires[0]] = obs_to_recipe_map[ob.name]
+
+            return word
+
+        if isinstance(observable, (qml.PauliX, qml.PauliY, qml.PauliZ)):
```

Comment:
Would it make sense to add identity here? I could in principle also have an observable with explicit `qml.Idendity`, right?

Reply:
You're right. I added cases for `qml.Identity` and a couple of tests.